File: back/src/models/enums.py
"""
Enums Python — correspondent 1:1 aux types PostgreSQL ENUM.
Utilisés dans les modèles SQLAlchemy et les schémas Pydantic.
"""

import enum


class UserRole(str, enum.Enum):
    """6 rôles définis dans le CDC §1.3"""
    student = "student"   # Élève cherchant un tuteur
    tutor   = "tutor"     # Tuteur proposant des sessions
    parent  = "parent"    # Parent d'un élève mineur
    admin   = "admin"     # Administrateur plateforme
    canope  = "canope"    # Acteur CANOPE
    cosp    = "cosp"      # Acteur COSP (hérite de CANOPE)


class TutorStatus(str, enum.Enum):
    """Cycle de validation d'un profil tuteur — CDC §MATCH-002"""
    pending   = "pending"    # En attente de review Admin/CANOPE
    validated = "validated"  # Visible publiquement
    rejected  = "rejected"   # Rejeté avec motif
    suspended = "suspended"  # Suspendu par Admin


class SessionStatus(str, enum.Enum):
    """Cycle de vie d'une session — CDC §RES"""
    pending_parent = "pending_parent"  # Attente validation parent (mineur)
    pending_tutor  = "pending_tutor"   # Attente acceptation tuteur
    confirmed      = "confirmed"       # Confirmée, Jitsi Meet généré
    in_progress    = "in_progress"     # En cours
    completed      = "completed"       # Terminée, escrow programmé
    cancelled      = "cancelled"       # Annulée
    disputed       = "disputed"        # Litige ouvert (Admin requis)


class PaymentStatus(str, enum.Enum):
    """Statuts d'un paiement Mobile Money — CDC §PAY"""
    pending    = "pending"
    processing = "processing"
    completed  = "completed"
    failed     = "failed"
    refunded   = "refunded"


class EscrowStatus(str, enum.Enum):
    """Statuts d'un fonds en séquestre — CDC §PAY"""
    held     = "held"      # Bloqué (session non terminée)
    released = "released"  # Libéré vers wallet tuteur
    disputed = "disputed"  # Litige gelé
    refunded = "refunded"  # Remboursé à l'élève/parent


class AssessmentStatus(str, enum.Enum):
    """Cycle d'un bilan d'orientation — CDC §COSP"""
    draft              = "draft"
    in_progress        = "in_progress"
    pending_validation = "pending_validation"  # Auto-bilan élève terminé, à relire par CANOPE/COSP
    validated          = "validated"  # Immuable sauf Admin


class MessageType(str, enum.Enum):
    """Types de messages — CDC §MSG"""
    text   = "text"
    image  = "image"
    file   = "file"
    system = "system"


class NotificationType(str, enum.Enum):
    """Types de notifications — CDC §NOTIF"""
    session_request = "session_request"
    payment         = "payment"
    message         = "message"
    review          = "review"
    system          = "system"


class ResourceType(str, enum.Enum):
    """Types de ressources CANOPE — CDC §CANOPE"""
    pdf      = "pdf"
    video    = "video"
    audio    = "audio"
    quiz     = "quiz"
    exercise = "exercise"


class ConversationType(str, enum.Enum):
    """Types de conversations — CDC §MSG"""
    student_tutor = "student_tutor"
    parent_tutor  = "parent_tutor"


class PaymentMethod(str, enum.Enum):
    """Méthodes de paiement acceptées — CDC §PAY"""
    mvola        = "mvola"
    orange_money = "orange_money"
    wallet       = "wallet"

File: front/types/enums.ts
/**
 * Enums TypeScript — miroir exact des types PostgreSQL ENUM.
 * Utilisés dans les types de tables, les composants et les appels API.
 * Toute modification ici doit être répercutée dans back/src/models/enums.py
 */

/** 6 rôles utilisateur définis dans le CDC §1.3 */
export enum UserRole {
  Student = "student",
  Tutor   = "tutor",
  Parent  = "parent",
  Admin   = "admin",
  Canope  = "canope",
  Cosp    = "cosp",
}

/** Cycle de validation d'un profil tuteur — CDC §MATCH-002 */
export enum TutorStatus {
  Pending   = "pending",
  Validated = "validated",
  Rejected  = "rejected",
  Suspended = "suspended",
}

/** Cycle de vie complet d'une session — CDC §RES */
export enum SessionStatus {
  PendingParent = "pending_parent",
  PendingTutor  = "pending_tutor",
  Confirmed     = "confirmed",
  InProgress    = "in_progress",
  Completed     = "completed",
  Cancelled     = "cancelled",
  Disputed      = "disputed",
}

/** Statuts d'une transaction Mobile Money — CDC §PAY */
export enum PaymentStatus {
  Pending    = "pending",
  Processing = "processing",
  Completed  = "completed",
  Failed     = "failed",
  Refunded   = "refunded",
}

/** Statuts d'un fonds en séquestre — CDC §PAY */
export enum EscrowStatus {
  Held     = "held",
  Released = "released",
  Disputed = "disputed",
  Refunded = "refunded",
}

/** Cycle d'un bilan d'orientation — CDC §COSP */
export enum AssessmentStatus {
  Draft             = "draft",
  InProgress        = "in_progress",
  PendingValidation = "pending_validation",
  Validated         = "validated",
}

/** Types de messages — CDC §MSG */
export enum MessageType {
  Text   = "text",
  Image  = "image",
  File   = "file",
  System = "system",
}

/** Types de notifications — CDC §NOTIF */
export enum NotificationType {
  SessionRequest = "session_request",
  Payment        = "payment",
  Message        = "message",
  Review         = "review",
  System         = "system",
}

/** Types de ressources pédagogiques CANOPE — CDC §CANOPE */
export enum ResourceType {
  Pdf      = "pdf",
  Video    = "video",
  Audio    = "audio",
  Quiz     = "quiz",
  Exercise = "exercise",
}

/** Types de conversations — CDC §MSG */
export enum ConversationType {
  StudentTutor = "student_tutor",
  ParentTutor  = "parent_tutor",
}

/** Méthodes de paiement Mobile Money — CDC §PAY */
export enum PaymentMethod {
  Mvola       = "mvola",
  OrangeMoney = "orange_money",
  Wallet      = "wallet",
}

/** Profils dominants des tests d'orientation */
export enum VakDominant   { V = "V", A = "A", K = "K" }
export enum DiscDominant  { D = "D", I = "I", S = "S", C = "C" }

/** Durées de session autorisées en minutes — CDC §RES */
export type SessionDuration = 60 | 90 | 120 | 180

/** Langues supportées — CDC §i18n */
export type SupportedLocale = "fr" | "mg"

/** Niveaux scolaires — CDC §4.1 student_profiles.grade_level */
export enum GradeLevel {
  Sixieme   = "6eme",
  Cinquieme = "5eme",
  Quatrieme = "4eme",
  Troisieme = "3eme",
  Seconde   = "2nde",
  Premiere  = "1ere",
  Terminale = "Tle",
  Superieur = "superieur",
}

/** Réactions emoji autorisées sur les messages */
export type AllowedEmoji = "👍" | "❤️" | "😂" | "😮" | "😢" | "🙏"
